src/thread.rs

```diff
@@ -299,7 +302,7 @@ impl<'a> Scope<'a> {
     /// scope exits.
     ///
     /// [spawn]: http://doc.rust-lang.org/std/thread/fn.spawn.html
-    pub fn spawn<F, T>(&self, f: F) -> ScopedJoinHandle<'a, T>
+    pub fn spawn<'s, F, T>(&'s self, f: F) -> ScopedJoinHandle<'s, T>
```

Comment:
`ScopedJoinHandle` must borrow the `Scope` (lifetime `'s`), not the environment outside the scope (lifetime `'a`).

Previously the following was compiling just fine:

```rust
let m = Mutex::new(None);
scope(|s| {
    let h = s.spawn(|| ());
    *m.lock().unwrap() = Some(h);
});
```